### PR TITLE
add filecoin-project oni as submodule and compile lotus-soup

### PR DESCRIPTION
**Pull request by nonsense**
This PR is adding `filecoin-project/oni` as a submodule to `lotus` and running a compilation step on every commit for `lotus-soup` test plan, so that we have the test plan in sync with the Lotus code base.

---

The idea of this PR is to make sure that `lotus-soup` test plan stays compatible with `filecoin-project/lotus` and we can run the test plan with a specific schedule (nightly, or every 6 hours, or whatever) based on `lotus` `master` branch.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -254,6 +254,25 @@ jobs:
...
+      Compile `lotus-soup` Testground test plan using the current version of Lotus.
+    parameters:
+      <<: *test-params
+    executor: << parameters.executor >>
+    steps:
+      - install-deps
+      - prepare
+      - run: cd extern/oni && git submodule sync
+      - run: cd extern/oni && git submodule update --init
+      - run: cd extern/oni/extra/filecoin-ffi && make
```
Comment (willscott) [inline .circleci/config.yml:268]: ℹ️ This is running the oni tests against the version of FFI that is submoduled in ONI. That will force lotus to update the oni FFI dep with the lotus one on updates. Potentially worth understanding if this is going to generate understandable errors if they get out of sync or if there's a better way to help us understand what went wrong

Comment (nonsense) [reply .circleci/config.yml:268]: Thanks for highlighting! We have to decide if this is a workflow that is okay from this repo or not.

Comment (magik6k) [reply .circleci/config.yml:268]: Ideally we'd use the submodule from lotus

File: .circleci/config.yml [hunk 2]
```diff
@@ -254,6 +254,25 @@ jobs:
           path: /tmp/test-reports
       - store_artifacts:
           path: /tmp/test-artifacts/conformance-coverage.html
+  test-lotus-soup:
```
Comment (magik6k) [inline .circleci/config.yml:257]: Looks like we didn't add this to any of the workflows at the bottom

Comment (nonsense) [reply .circleci/config.yml:257]: Good catch. Somehow I missed copying this from the test project.

File: .circleci/config.yml (hunk 1 above)
Comment (nonsense) [reply .circleci/config.yml]: I've added another `go mod replace` so that we use the submodule of `filecoin-ffi` from the `lotus` repo.

**Review by magik6k — APPROVED** [review]

**Pull request merged**